### PR TITLE
Rename Flags EventFlag and MemFdCreateFlag

### PR DESCRIPTION
**Pull request by howjmay**
relates to #317

## What does this PR do

Rename Flags 

## Checklist:

- [x] I have read `CONTRIBUTING.md`
- [ ] I have written necessary tests and rustdoc comments
- [x] A change log has been added if this PR modifies nix's API

### CODE REVIEW

**Review by SteveLauC — COMMENTED** [review]
Existing changes look good, thanks!

Though you need to address the CI failures, i.e., rename the usages on other platforms as well.

And can we have something like this to help users migrate:

```rs
#[deprecated(since = "0.30.0", note = "Use `EvFlags instead`")]
pub type EventFlag = EvFlags;
```

**Comment by SteveLauC** [comment]
Also, a changelog entry is needed, see [this](https://github.com/nix-rust/nix/blob/master/CONTRIBUTING.md#pull-requests) on how to add one.

File: test/sys/test_event.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ fn test_struct_kevent() {
     let actual = KEvent::new(
         0xdead_beef,
         EventFilter::EVFILT_READ,
```
Comment (SteveLauC) [inline test/sys/test_event.rs:13]: @asomers, any idea what we should do to things like this `enum EventFilter`? 

```
use nix::sys::event::{EvFlags, EventFilter};
```

Honestly, it feels weird to me when putting them together 🥲, one aims to be short, typical C-style, one aims to be self-explanatory.

**Review by SteveLauC — APPROVED** [review]
Looks good, thank you:)

File: test/sys/test_event.rs (hunk 1 above)
Comment (asomers) [reply test/sys/test_event.rs:13]: If does seem odd now. I think they'll need hard to remember like this. We should probably make the two names more consistent.

Comment (SteveLauC) [reply test/sys/test_event.rs:13]: Yeah

**Comment by howjmay** [comment]
thank you too

**Comment by SteveLauC** [comment]
Let's merge this first.

**Pull request merged**